### DISCUSSION TITLE
How to redirect to previous page when using signInWithOAuth and @supabase/ssr?

### DISCUSSION THREAD

**Discussion by wong2** [supabase/supabase#21441] [Questions]
I'm following this guide [OAuth with PKCE flow for SSR
](https://supabase.com/docs/guides/auth/server-side/oauth-with-pkce-flow-for-ssr?framework=remix), it's working, but I can't find how to redirect users back to previous page after they log in.

What I want: `Page A => Login page => Page A`, current: `Page A => Login page => Home page`

**Comment by mwoss** [comment]
Have you added `Page A` to permitted redirect URLs in the Supabase dashboard? If not you can do it at `Dashboard -> Authentication -> URL configuration -> Redirect URLs`. This setup is required for security reasons.

**Reply by wong2** [reply]
I think you misunderstood the question

**Reply by mwoss** [reply]
I thought that `redirectTo` parameter is not working for you and users are getting redirected to the home page (site URL). Is it not the problem here?

**Reply by wong2** [reply]
No, please read this guide first: https://supabase.com/docs/guides/auth/server-side/oauth-with-pkce-flow-for-ssr?framework=remix

**Comment by silentworks** [comment]
The link you are pointing to has code in its source to handle that, this isn't Supabase specific but just general web development workings. If you add a query string called `next` to the `redirectTo` url of the `signInWithOAuth` call it will redirect to that url. You can either make the path a full path or just the relative path. 

> Do note there is a security concern around this way of doing things since someone could use your website as a phishing tool and redirect users to a different domain all together. You should put some checks in place to make sure the url from the `next` query string is to a trusted domain (your own domain that is).

```ts
await supabase.auth.signInWithOAuth({
  provider,
  options: {
    redirectTo: `http://example.com/auth/callback?next=/page-a`,
  },
})
```
